JOSESwift/Sources/JWEHeader.swift

```diff
@@ -104,6 +104,13 @@ public extension JWEHeader {
         // swiftlint:disable:next force_cast
         return SymmetricKeyAlgorithm(rawValue: parameters["enc"] as! String)
     }
+    
+    public var compressionAlgorithm: CompressionAlgorithm? {
+        guard let compressionAlgorithm = parameters["zip"] as? String else {
+            return nil
```

Comment:
We could return `NONE` here. `.NONE` gives us the ability to differentiate between no algorithm and an unknown algorithm (`nil` returned by the raw value initializer below).

The [RFC](https://tools.ietf.org/html/rfc7516#section-4.1.3) says: 

> If no "zip" parameter is present, no compression is applied
   to the plaintext before encryption.
> This Header Parameter MUST be understood and processed by
   implementations.

It is a bit unclear, but by using `.NONE` when the parameter is absent and marking an unknown value as `nil` we should be on the safe side in my opinion.